seaport-subgraph/src/seaport.ts

```diff
@@ -0,0 +1,220 @@
+                "txHash consideration log tx {} logindex {} paymentToken {} paymentAmount {} recipient {} buyer {} seller {} index {}",
+                [
+                    txHash.toHexString(),
+                    event.logIndex.toString(),
+                    paymentToken.toHexString(),
+                    paymentAmount.toString(),
+                    consideration.recipient.toHexString(),
+                    buyer.toHexString(),
+                    seller.toHexString(),
+                    i.toString(),
+                ]
+            );
+        } else {
+          if (buyer == Address.zero()){
+            buyer = consideration.recipient;
+          }
+          let standard = isERC721(consideration.itemType)
+            ? NftStandard.ERC721
+            : isERC1155(consideration.itemType)
+            ? NftStandard.ERC1155
+            : NftStandard.UNKNOWN;
+          let nft = new airstack.nft.NFT(consideration.token, standard, consideration.identifier, consideration.amount);
+
+          let sale = new airstack.nft.Sale(buyer, seller, nft, paymentAmount, paymentToken, protocolFees, protocolBeneficiary, new Array<airstack.nft.CreatorRoyalty>());
+          allSales.push(sale);
+          
+          log.info(
+            "txHash consideration log tx {} logindex {} nftContract {} NFTId {} index {} recipient {} buyer {} seller {}",
+            [
+              txHash.toHexString(),
+              event.logIndex.toString(),
+              consideration.token.toHexString(),
+              consideration.identifier.toString(),
+              i.toString(),
+              consideration.recipient.toHexString(),
+              buyer.toHexString(),
+              seller.toHexString(),
+            ]
+          );
+        }
+    }
+    
+    for(let i = 0; i <allSales.length; i++){
+        let royalty = new airstack.nft.CreatorRoyalty(royaltyFees.div(BigInt.fromI64(allSales.length)), royaltyBeneficiary);
+        allSales[i].royalties.push(royalty);
+        allSales[i].protocolFees = protocolFees.div(BigInt.fromI64(allSales.length));
+        allSales[i].protocolFeesBeneficiary = protocolBeneficiary;
+        allSales[i].paymentAmount = paymentAmount.div(
+          BigInt.fromI32(allSales.length)
+        ); // For bundle sale, equally divide the payment amount in all sale transaction
+        allSales[i].paymentToken = paymentToken;
+        log.info("txHash {} royaltyBeneficiary {} feeBeneficiary {} feeAmount {}",
+          [ 
+            txHash.toHexString(),
+            royaltyBeneficiary.toHexString(),
+            protocolBeneficiary.toHexString(),
+            // allSales[i].royalties[0].beneficiary.toHexString(),
+            allSales[i].protocolFees.toString(),
+          ]
+        )
+    }
+    log.info("txHash {} allSales length {}", [txHash.toHexString(), allSales.length.toString()]);
+    if (allSales.length > 0) {
+      if (allSales[0].seller == Address.zero() || allSales[0].buyer == Address.zero()){
```

Comment:
Why 0th index recored is used?

Reply:
Refactored the code, now checking it for ith index